llrt_core/src/modules/http/request.rs

```diff
@@ -60,6 +61,7 @@ impl<'js> Request<'js> {
             method: "GET".to_string(),
             headers: None,
             body: None,
+            content_type: None,
```

Comment:
Thanks for the review.
Your comment was difficult for me to understand(All this is due to my lack of English skills...), so I modified it as follows, while also checking the behavior of Bun.

1. If both blob's type and Headers' content-type are specified, blob's type is used.
2. If blob's type is not specified and Headers' content-type is specified, Headers' content-type is used.
3. If both are not specified, nothing is specified.

Reply:
Sorry. I forgot one more thing...

4. if the type of Blob is specified and the content-type of Headers is not specified, the content-type of Headers should be the type of Blob.

Have you been told about this?

Reply:
1. No, it looks like the header value always sets the blobs value. If no is specified, it's just an empty type (even if blob as a type defined. Test this in Chrome:
```javascript
let b = new Blob(["test"],{type:"application/json"})
let r = new Request(b,{headers: {"content-type":"text/plain"}})

await r.blob();

let b = new Blob(["test"],{type:"application/json"})
let r = new Request(b)

await r.blob();

```
So if content-type header is present, it will be used when calling `blob()` otherwise blob will have empty type (even if specified when constructed)

This basically means that we don't need a `content_type` field, it's never used or derived from blob instance

2. See 1
3. Yes
4. No, headers are not touched.

Reply:
Indeed!
I see that the blob’s type is not used in both of the following patterns.

1.
<img width="379" alt="image" src="https://github.com/awslabs/llrt/assets/7500514/bb39788c-f072-49f9-814f-5f14705f05f6">

2.
<img width="368" alt="image" src="https://github.com/awslabs/llrt/assets/7500514/d1aaf780-fbd1-41b5-b1d2-f58d231c3de3">

This way, there is no need to have a separate content-type inside the request object.
So we can just set the content-type of headers to the type used in blob(), regardless of whether it has a value or not.

Reply:
I was wondering if it would be possible to simplify the method of retrieving the content-type a bit more, but I didn't have any good ideas. I have made the fix and would appreciate a review.

Reply:
The only other way I can think of is this:
```rust
let mime_type = headers.iter()
    .find_map(|(k, v)| if k == "content-type" { Some(v.clone()) } else { None });
```

Reply:
Thank you very much. I think I have now addressed all the issues.